docs/extensions/ca-rotation.md

```diff
@@ -0,0 +1,76 @@
+# CA Rotation in Extensions
+
+[GEP-18](../proposals/18-shoot-CA-rotation.md) adds support for automated rotation of Shoot cluster certificate authorities (CAs).
+This document outlines all requirements, that Gardener extensions need to fulfill in order to support the CA rotation feature.
+
+## Requirements for Shoot Cluster CA Rotation
+
+- Extensions must not rely on static CA `Secret` names managed by gardenlet, because their names are changing during CA rotation.
+- Extensions cannot issue or use client certificates for authenticating against shoot API servers. Instead, they should use short-lived auto-rotated `ServiceAccount` tokens via gardener-resource-manager's `TokenRequestor`. Also see [Conventions](./conventions.md) and [`TokenRequestor`](../concepts/resource-manager.md#tokenrequestor) documents.
+- Extensions need to generate dedicated CAs for signing server certificates (e.g. `cloud-controller-manager`). There should be one CA per controller and purpose in order to bind the lifecycle to the reconciliation cycle of the respective object for which it is created.
+- CAs managed by extensions should be rotated in lock-step with the shoot cluster CA. 
+  When the user triggers a rotation, gardenlet writes phase and initiation time to `Shoot.status.credentials.rotation.certificateAuthorities.{phase,lastInitiationTime}`. See [GEP-18](../proposals/18-shoot-CA-rotation.md#rotation-sequence-for-cluster-and-client-ca) for a detailed description on what needs to happen in each phase.
+  Extensions can retrieve this information from [`Cluster.shoot.status`](./cluster.md).
+
+## Utilities for Secrets Management
+
+In order to fulfill the requirements listed above, extension controllers can reuse the [`SecretsManager`](../development/secrets_management.md), that gardenlet uses to manage all shoot cluster CAs, certificates and other secrets as well.
+It implements the core logic for managing secrets that need to be rotated, auto-renewed etc.
+
+Additionally, there are utilities for reusing `SecretsManager` in extension controllers.
+They already implement above requirements based on the `Cluster` resource and allow focusing on the extension controllers' business logic.
+
+For example, a simple `SecretsManager` usage in an extension controller could look like this:
+
+```go
+// identity for SecretsManager instance in ControlPlane controller
+const identity = "provider-foo-controlplane"
+
+func Reconcile() {
+  var (
+    cluster *extensionscontroller.Cluster
+    client  client.Client
+
+    // define wanted secrets with options
+    secretConfigs = []extensionssecretsmanager.SecretConfigWithOptions{
+      {
+        // dedicated CA for ControlPlane controller
+        Config: &secretutils.CertificateSecretConfig{
+          Name:       "ca-provider-foo-controlplane",
```

Comment:
Nit: Extract the name in a variable that can be re-used in the `SignedByCA` option to prevent duplication